### PR TITLE
feat(focuslock): add focusLock component

### PR DESCRIPTION
**Pull request by jamesb3ll**
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/161)

### CODE REVIEW

**Comment by rachmulvey** [comment]
Is this one still needed?

File: src/components/FocusLock/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { Component, Children } from 'react';
+import { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+
+// From: https://github.com/davidtheclark/tabbable
+const tabbableSelector = [
```
Comment (tysonwolker) [inline src/components/FocusLock/index.js]: This doesn't take into account hidden elements and tabindex

Comment (tysonwolker) [reply src/components/FocusLock/index.js]: Also input[type=hidden] needs to be excluded too.

Comment (tysonwolker) [reply src/components/FocusLock/index.js]: And non native elements ones that have tab-index added for example

**Pull request closed without merging**